processing/src/main/java/io/druid/query/filter/BoundDimFilter.java

```diff
@@ -385,19 +385,20 @@ private void initLongData()
           if (hasLowerBound()) {
             final Long lowerLong = GuavaUtils.tryParseLong(lower);
             if (lowerLong == null) {
-              matchesAnything = false;
-              return;
+              // Unparseable values fall before all actual numbers, so all numbers will match the lower bound.
+              hasLowerLongBoundVolatile = false;
+            } else {
+              hasLowerLongBoundVolatile = true;
+              lowerLongBoundVolatile = lowerLong;
             }
-
-            hasLowerLongBoundVolatile = true;
-            lowerLongBoundVolatile = lowerLong;
           } else {
             hasLowerLongBoundVolatile = false;
```

Comment:
@himanshug having or not having `volatile` on all other fields doesn't make any difference to "safety" or guarantees provided, so if @gianm chose to add a comment, it probably should have been here before as well.

Also having `volatile` on all other fields doesn't protect you from introducing subtle bugs as described, E. g. reordering reads or writes would break this code despite all fields are `volatile`.